cloudmock/aws/mockec2/launch_templates.go

```diff
@@ -141,6 +141,10 @@ func (m *MockEC2) CreateLaunchTemplate(request *ec2.CreateLaunchTemplateInput) (
 		SecurityGroupIds:      request.LaunchTemplateData.SecurityGroupIds,
 		SecurityGroups:        request.LaunchTemplateData.SecurityGroups,
 		UserData:              request.LaunchTemplateData.UserData,
+		MetadataOptions: &ec2.LaunchTemplateInstanceMetadataOptions{
+			HttpTokens:              request.LaunchTemplateData.MetadataOptions.HttpTokens,
+			HttpPutResponseHopLimit: request.LaunchTemplateData.MetadataOptions.HttpPutResponseHopLimit,
+		},
```

Comment:
Please move below and check for `request.LaunchTemplateData.MetadataOptions != nil`.

Reply:
Done